src/Query/ConditionGroup.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Drupal\entity\Query;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheableDependencyInterface;
+use Drupal\Core\Cache\RefinableCacheableDependencyTrait;
+
+/**
+ * Represents a group of access conditions.
+ *
+ * Used by query access handlers for filtering lists of entities based on
+ * granted permissions.
+ *
+ * Examples:
+ * @code
+ *   // Filter by node type and uid.
+ *   $condition_group = new ConditionGroup();
+ *   $condition_group->addCondition('type', ['article', 'page']);
+ *   $condition_group->addCondition('uid', '1');
+ *
+ *   // Filter by node type or status.
+ *   $condition_group = new ConditionGroup('OR');
+ *   $condition_group->addCondition('type', ['article', 'page']);
+ *   $condition_group->addCondition('status', '1', '<>');
+ *
+ *   // Nested condition groups: node type AND (uid OR status).
+ *   $condition_group = new ConditionGroup();
+ *   $condition_group->addCondition('type', ['article', 'page']);
+ *   $condition_group->addCondition((new ConditionGroup('OR'))
+ *     ->addCondition('uid', 1)
+ *     ->addCondition('status', '1')
+ *   );
+ * @endcode
+ */
+final class ConditionGroup implements \Countable, CacheableDependencyInterface {
+
+  use RefinableCacheableDependencyTrait;
```

Comment:
This implements `CacheableDependencyInterface`, so `RefinableCacheableDependencyTrait` is wrong, because it adds mutability methods. You want `CacheableDependencyTrait` instead.